torch/fx/experimental/proxy_tensor.py

```diff
@@ -2500,8 +2553,49 @@ def handle_sym_dispatch(
 
 
 @contextmanager
-def disable_proxy_modes_tracing() -> Generator[ProxyTorchDispatchMode, None, None]:
-    return _disable_infra_mode(torch._C._TorchDispatchModeKey.PROXY)
+def disable_proxy_modes_tracing(
+    trace_symbolics: bool = False,
+) -> Generator[ProxyTorchDispatchMode, None, None]:
+    """
+    Temporarily disable the current PROXY dispatcher.
+
+    If `trace_symbolics` is True then disables dispatch tracing but continues
+    tracing symbolic expressions (like `s1*s2`).
+
+    Why might you need this?  Let's say you have some code which is tracing an
+    operation and in the middle needs to do some FakeTensor operations to
+    compute a size but it doesn't want those temporary operations to be added to
+    the graph - so it disables tracing. If it uses the result of those
+    operations and they contain a SymNode then the proxy will not know the
+    provinance of the SymNode and will raise an exception.
+
+    Be careful when using this because the automatic disabling of the dispatch
+    mode that you get won't apply to this override!
+    """
+    saved_sym_mode_override = ProxyTorchDispatchMode._global_state.sym_mode_override
+    mode_unset = _unset_infra_mode(torch._C._TorchDispatchModeKey.PROXY)
+    if trace_symbolics:
+        # Make sure that if they already override the mode we use that, not just
+        # blindly set from the current mode.
+        #
+        # TODO: What's the right thing to do if we have a nesting like this?
+        # Currently does NOT trace syms in the inner code. Do we need to
+        # remember the "last" dispatch mode too?
+        #
+        #     disable:
+        #         disable(trace=True):
+        #             ...
+        #
+        new_sym_mode_override = saved_sym_mode_override or mode_unset
```

Comment:
would we ever expect case like this? 

```
new_proxy_mode:
  disable:
    new_proxy_mode:
      disable:
```

should we add some defensive asserts here for these edge cases we don't expect to support?